### DISCUSSION TITLE
Hardware assembly

### DISCUSSION THREAD

**Discussion by ronkmd** [HASwitchPlate/openHASP#165] [Hardware]
Hi. This project looks really interesting. Is there any more info on the actual hardware assembly? I'm particularly interested in US single gang switch plate. I see there's some info on the original HASPone assembly but I'm wondering what the differences are when using openHASP hardware. I'm really a newbie at this. Done a couple of ESP8266 projects already (ESPLink, Shelly, ESPHome). PS: I did order an M5Stack Core 2.

**Comment by fvanroie** [comment]
Hi, we're still mainly focusing on the firmware aspect at the moment.

The hardware is a lot more diverse and not limited to the D1-mini and Nextion displays. So there are a lot of possible hardware combinations. I don't know of any readily available US single gang enclosures at this stage... We'll certainly list them when they become available.

**Comment by hsmalley** [comment]
Is it possible to to run openhasp on the haspone some time in the future?

**Reply by fvanroie** [reply]
While the purpose is the same, the hardware for both projects is different, so you have to use the firmware according to the devices you have:
- **HASPone:** D1-mini ESP8266 with Nextion display
- **openHASP**: ESP32 with commodity display

openHASP needs direct control of the display, which Nextion doesn't provide. So that means you cannot run openHASP on HASPone hardware.

**Comment by nickrout** [comment]
There is a person on the home assistant forum who has put a Wireless-Tag WT32-SC01 on top of a Sonoff T1. The discussion starts here https://community.home-assistant.io/t/openhasp-an-mqtt-driven-touchscreen-scene-controller/300853/53?u=nickrout. This will fit a US single gang box, which are pretty well the same size as those in NZ.
